src/DSL/TSParameter.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// 
+
+namespace AutoRest.TypeScript.DSL
+{
+    /// <summary>
+    /// A data class that contains the details needed to describe a TypeScript parameter.
+    /// </summary>
+    public class TSParameter
```

Comment:
I think you should reconsider this path unless you're positive it's the way you want-- it's starting to get beyond "fancy stringbuilders" toward "a data model you have to contend with and maintain along with the AutoRest model"-- I took away from our Java experience that the first thing is excellent while the second thing winds up being harder to maintain and understand.

Reply:
Figuring out "why is the property value this way in the TSParameter" means traversing additional indirections compared to using the AutoRest Parameter type directly with the TSBuilder.

Reply:
I tried to remove this and found very quickly that it was not as easy as I thought it would be. We have a lot of special logic around converting the Parameter's IModelType to its generated string representation that made it cumbersome for me to say, "This is the type I want to generate."
I understand your apprehension at seeing me introduce things that remind you of the work I did in the Java generator. I have definitely learned where I went wrong in that situation, and I'd be happy to talk to you about how this is a different scenario. I don't think a PR is the best place to have that conversation.
TLDR: I'm going to leave TSParameter as I think it helps the generation scenario, but I do appreciate the reminder/warning that my efforts in the Java generator didn't end up as well as I had hoped. :)